giskard/scanner/llm/prompt_injection/evaluator.py

```diff
@@ -0,0 +1,59 @@
+import re
+import string
+from abc import ABC, abstractmethod
+from colorama import Fore, Style
+
+from ...scanner import logger
+from .data import Prompt, StringMatchingMethod
+
+
+class Evaluator(ABC):
+    """Base class for evaluators that define a way of detecting a LLM failure"""
+
+    def __init__(self,
+                 name: str = None):
+        if name is None:
+            self.name = __class__
+
+        print(f"loading {Style.RESET_ALL}{Fore.LIGHTBLUE_EX}Evaluator{Style.RESET_ALL}: {self.name}")
+        logger.info(f"Evaluator init: {self}")
+
+    @abstractmethod
+    def evaluate(self, prediction: str, prompt: Prompt) -> bool:
+        """Takes a prediction and a prompt and returns 1 if vulnerability is found."""
+        raise NotImplementedError
+
+
+class StringMatcher(Evaluator):
+
+    @staticmethod
+    def evaluate(prediction: str, prompt: Prompt) -> bool:
```

Comment:
`scanner.llm.prompt_injection.evaluator.Evaluator.evaluate` should also be static